arboretum/ibm_cloud/fetchers/fetch_cluster_resource.py

```diff
@@ -0,0 +1,154 @@
+# -*- mode:python; coding:utf-8 -*-
+# Copyright (c) 2021 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""IBM Cloud cluster resource fetcher."""
+
+import io
+import json
+import os
+import pathlib
+import tempfile
+import zipfile
+
+from arboretum.common.iam_ibm_utils import get_tokens
+from arboretum.common.kube_constants import RESOURCE_TYPES_DEFAULT
+from arboretum.common.kube_utils import get_cluster_resources
+
+from compliance.evidence import (
+    DAY, RawEvidence, get_evidence_dependency, store_raw_evidence
+)
+from compliance.fetch import ComplianceFetcher
+
+import yaml
+
+
+class ICClusterResourceFetcher(ComplianceFetcher):
+    """Fetch resources of IBM Cloud Kubernetes clusters."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Initialize the fetcher object with configuration settings."""
+        cls.config.add_evidences(
+            [
+                RawEvidence(
+                    'cluster_resources.json',
+                    'ibm_cloud',
+                    DAY,
+                    'IBM Cloud Kubernetes cluster resources'
+                )
+            ]
+        )
+        cls.resource_types = cls.config.get(
+            'org.ibm_cloud.cluster_resources.types', RESOURCE_TYPES_DEFAULT
+        )
+        cls.tempdir = tempfile.TemporaryDirectory()
+        return cls
+
+    @classmethod
+    def tearDownClass(cls):
+        """Cleanup class."""
+        cls.tempdir.cleanup()
+
+    def _get_iks_credentials(self, cluster_config):
+        """Get credentials for an IKS cluster.
+
+        This function implements the procedure described in
+        https://cloud.ibm.com/apidocs/kubernetes#getclusterconfig
+        """
+        for name in cluster_config.namelist():
+            p = pathlib.Path(name)
+            if p.name.startswith('kube-config'):
+                kubeconfig = yaml.safe_load(cluster_config.read(name))
+                cluster_token = kubeconfig['users'][0]['user'][
+                    'auth-provider']['config']['id-token']
```

Comment:
Please fix this.  Why not:

```suggestion
                usr = kubeconfig['users'][0]['user']
                cluster_token = usr['auth-provider']['config']['id-token']
```